rfcs/0000-policy_target_scope.md

```diff
@@ -0,0 +1,267 @@
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: foo-route
+spec:
+  parentRefs:
+  - name: my-gw
+  hostnames:
+  - "example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /foo
+    backendRefs:
+    - name: foo
+      port: 8080
+```
+
+With `foo-route` declaring a single hostname (`example.com`) and a single `HTTPRouteMatch` (anything
+starting with `/foo`), the policy will apply to the same traffic: `example.com/foo*`, so that `foo` 's
+traffic to port `8080` will be painted blue, as specified by the `default` section of our `PaintPolicy`.
+
+In order to change that color, we'd need to define another `PaintPolicy` that defines a `default` and
+targets our `foo-route`:
+
+```yaml
+kind: PaintPolicy
+spec:
+  default:
+    color: green
+  targetRef:
+    kind: HTTPRoute
+    name: foo-route
+```
+
+Attaching the `PaintPolicy` above to our `HTTPRoute` will _shadow_ the `PaintPolicy` attached to the
+`Gateway`. It is important to note that there is no way for the policy to discriminate the traffic it
+applies to at this stage. All traffic to `foo-route` is now painted `green`, while any further routes
+attached to `my-gw` would still be painted `blue`.
+
+>**Note:** The status block of a policy should reflect whether it is shadowing or being shadowed by other
+>policies at that level of the hierarchy.
+
+Which would effectively disable all rate limiting traffic to the `foo-route` and ultimately the `foo`
+`Backend`.
+
+Using another example, let's consider a single `HTTPRoute` with two `HTTPRouteRule`s:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: bar-route
+spec:
+  parentRefs:
+  - name: my-gw
+  hostnames:
+  - "bar.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+    backendRefs:
+    - name: bar-svc
+      port: 8081
+  - matches:
+    - headers:
+      - type: Exact
+        name: env
+        value: production
+    backendRefs:
+    - name: bar-svc
+      port: 8080
+```
+
+It is important to consider that only _one_ rule will ever be match for any given requests. It makes for
+a simpler model to have only one _Policy_ to also only ever being matched, aligning the both behaviors.
+To apply a policy on the first rule only, the user needs to duplicate that match on the policy targeting
+`bar-route`:
+
+```yaml
+kind: PaintPolicy
+spec:
+  default:
+    color: red
+  targetRef:
+    kind: HTTPRoute
+    name: foo-route
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+```
+
+Hereby we are coloring the traffic to our `/login` on port `8081` to be `red`. That coloring _will not_
+apply to other matches. So that traffic to `bar-svc` on port `8080` would be colored `green` as per the
+`PaintPolicy` attached to `my-gw`.
+
+The resolving of _Policies_ to enforce occurs at the `HTTPRoute` or within individual `HTTPRouteRule`
```

Comment:
I like that we can describe this relatively simply. That said I thnk we would want a good range of "example" policies to cover common use cases similar to how it is done for https://github.com/kubernetes-sigs/gateway-api/tree/main/examples/standard